test/nodejs/test_file.rb

```diff
@@ -147,15 +147,21 @@ def test_windows_separators
   end if windows_platform?
 
   def test_windows_file_expand_path
-    assert_equal(Dir.pwd.gsub(/\\/, '/') + '/foo/bar.js', File.expand_path('./foo/bar.js'))
-    assert_equal('/foo/bar.js', File.expand_path('/foo/bar.js'))
+    drive_letter = Dir.pwd.slice(0, 2)
+    assert_equal(Dir.pwd + '/foo/bar.js', File.expand_path('./foo/bar.js'))
+    assert_equal(drive_letter + '/foo/bar.js', File.expand_path('/foo/bar.js'))
     assert_equal('c:/foo/bar.js', File.expand_path('c:/foo/bar.js'))
     assert_equal('c:/foo/bar.js', File.expand_path('c:\\foo\\bar.js'))
-    assert_equal('c:/foo/bar.js', File.expand_path( 'bar.js', 'c:\\foo'))
-    assert_equal('c:/foo/baz/bar.js', File.expand_path('\\baz\\bar.js', 'c:\\foo'))
-    assert_equal('c:/baz/bar.js', File.expand_path( '\\baz\\bar.js', 'c:\\foo\\..'))
-    assert_equal('c:/foo/bar.js', File.expand_path( '\\..\\bar.js', 'c:\\foo\\baz'))
-    assert_equal('c:/foo/bar.js', File.expand_path( '\\..\\bar.js', 'c:\\foo\\baz\\'))
+    assert_equal('c:/foo/bar.js', File.expand_path('bar.js', 'c:\\foo'))
```

Comment:
Some assertions were wrong.
For instance `\\baz\\bar.js` is an absolute path, so `File.expand_path( '\\baz\\bar.js', 'c:\\foo')` should be equal to `c:/baz/bar.js` (not `c:/foo/baz/bar.js`).

Reply:
👍 maybe check with ruby spec's code, if you didn't already

Reply:
I did check and they are also losing their sanity: `# FIXME: these are insane!`

https://github.com/ruby/ruby/blob/62ebf35592717655a34620301fd45dfc522af1e2/spec/ruby/core/file/expand_path_spec.rb#L94-L101

We could probably add (carefully) more tests cases but I think for now we are good :sweat_smile: